spi/control-plane/contract-spi/src/main/java/org/eclipse/edc/connector/contract/spi/offer/ConsumerOfferResolver.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.contract.spi.offer;
+
+import org.eclipse.edc.connector.contract.spi.validation.ContractValidationService;
+import org.eclipse.edc.connector.contract.spi.validation.ValidatableConsumerOffer;
+import org.eclipse.edc.spi.result.ServiceResult;
+import org.eclipse.edc.spi.types.domain.offer.ContractOffer;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Resolve the consumer offer into a {@link ValidatableConsumerOffer} which can be used
+ * to validate incoming offer through {@link ContractValidationService#validateInitialOffer}
+ */
+public interface ConsumerOfferResolver {
+
+    @NotNull
+    ServiceResult<ValidatableConsumerOffer> resolveOffer(ContractOffer offer);
```

Comment:
I'd leave only the second method, and let the caller extract the id from the offer to call the `resolveOffer` method